test/k8sT/CustomCalls.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sTest
+
+import (
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/cilium/cilium/pkg/defaults"
+	. "github.com/cilium/cilium/test/ginkgo-ext"
+	"github.com/cilium/cilium/test/helpers"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("K8sCustomCalls", func() {
+
+	var (
+		kubectl *helpers.Kubectl
+	)
+
+	type customCallDirection uint32
+
+	const (
+		// Constants for tail call hooks
+		// See CUSTOM_CALLS_IDX_* defines in bpf/lib/maps.h
+		IngressIPv4 customCallDirection = 0
+		EgressIPv4  customCallDirection = 1
+		IngressIPv6 customCallDirection = 2
+		EgressIPv6  customCallDirection = 3
+		// eBPF virtual file system
+		bpffsDir string = defaults.DefaultMapRoot + "/" + defaults.DefaultMapPrefix + "/"
+	)
+
+	BeforeAll(func() {
+		kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+		deploymentManager.SetKubectl(kubectl)
+	})
+
+	AfterEach(func() {
+		deploymentManager.DeleteAll()
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport("cilium status", "cilium endpoint list")
```

Comment:
Would other bpftool outputs be useful here? E.g., maps, progs content of new tail call map.

Reply:
Possibly. I'll look at this, and at the potential bugtool additions, as a follow-up.